src/GraphQL/Attributes/ParserAttribute.cs

```diff
@@ -0,0 +1,91 @@
+using System.Reflection;
+using GraphQL.Types;
+
+namespace GraphQL;
+
+/// <summary>
+/// Specifies a custom parser method for a field argument or a field of an input object in a GraphQL schema.
+/// </summary>
+[AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Property | AttributeTargets.Field)]
+public sealed class ParserAttribute : GraphQLAttribute
+{
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+    private readonly Type? _parserType;
+    private readonly string _parserMethodName;
+    private readonly bool _includePrivate;
+
+    /// <summary>
+    /// Specifies a custom parser method for a field of an input object in a GraphQL schema using the specified parser method name.
+    /// The method must exist on the declaring type of the field and be declared as static.
+    /// The method may be private or public and must have the signature 'object <paramref name="parserMethodName"/>(object value)'.
+    /// If it is public, you may wish to mark the method with <see cref="IgnoreAttribute"/> to prevent it from being exposed in the schema.
+    /// </summary>
+    [RequiresUnreferencedCode("Please ensure the specified method is not trimmed or use an alternative constructor.")]
+    public ParserAttribute(string parserMethodName)
+    {
+        _parserMethodName = parserMethodName
+            ?? throw new ArgumentNullException(nameof(parserMethodName));
+        _includePrivate = true;
```

Comment:
Why not allow `protected` as well? Also, allowing `internal` when the type is specified sounds useful.

Reply:
BindingFlags are public and non public. This statement actually allows all non public including protected and internal. 

Normally a reference like this would only reference public members, but since with type-first public members are exposed through GraphQL, I allowed non public members for this constructor. The other constructors can refer to internal or private classes but the method must be declared as public.